### ISSUE TITLE
istioctl kube-inject readiness etc options no longer respected

### ISSUE THREAD

**Issue by esnible** [istio/istio#12078]
https://preliminary.istio.io/docs/reference/commands/istioctl/#istioctl-kube-inject documents the following

--readinessFailureThreshold <uint32>
--readinessInitialDelaySeconds <uint32>
--readinessPeriodSeconds <uint32>
--rewriteAppProbe
--statusPort

These no longer work.  Example:

```
istioctl kube-inject --readinessFailureThreshold 1234 -f samples/bookinfo/platform/kube/bookinfo.yaml | grep failureThreshold
          failureThreshold: 30
          failureThreshold: 30
          failureThreshold: 30
          failureThreshold: 30
          failureThreshold: 30
          failureThreshold: 30
```

Perhaps this is due to the way the Configmap is now structured.  There are annotations, eg.

```
[[ annotation .ObjectMeta `readiness.status.sidecar.istio.io/failureThreshold`
```

Perhaps the CLI feature broke when the annotation names were set.

**Comment by ayj** [comment]
A few things to note here:

1) There is an stray `-f` before `--readinessFailureThreshold` which affects how the flag is processed.

2) Many of the command line flags are deprecated in favor of using the template file via --injectConfigMapName or --injectConfigFile). It looks like --readinessFailureThreshold and some of the other newer flags were not added to this deprecated list. 
 https://github.com/istio/istio/blob/f8923d594af5bde2f3b63020d95436c2916946fa/istioctl/cmd/istioctl/kubeinject.go#L414-L418).

3) It looks like the hardcoded template and parameters is being removed in master (see https://github.com/istio/istio/pull/10830).

**Comment by ayj** [comment]
With regards to the flag not taking affect: none of the flag overrides will take affect unless `--hub` or `--tag` are specified.

**Comment by esnible** [comment]
The extra `-f` was my mistake in entering the Issue.  I have removed it.

If I supply `--hub` I get a seemingly incorrect response:

```
istioctl kube-inject --readinessFailureThreshold 1234 -f samples/bookinfo/platform/kube/bookinfo.yaml --hub test --injectConfigMapName=istio-sidecar-injector
Flag --hub has been deprecated, Use --injectConfigMapName or --injectConfigFile instead
Error: one of injectConfigFile or injectConfigMapName is required
```

Same incorrect (?) response if I supply `--injectConfigFile=/tmp/injectConfigFile.yaml`.

**Comment by ayj** [comment]
cc @mandarjog 

The flags are deprecated. I believe they were only retained while our test framework updated. You'll also need to specify ISTIOCTL_USE_BUILTIN_DEFAULTS=1 and set the --tag as well. 

```
ISTIOCTL_USE_BUILTIN_DEFAULTS=1 ~/go/bin/istioctl kube-inject \
    --coreDump 1234 \ 
    -f nginx-deployment.yaml \
    --hub foo \
    --tag bar
```

**Comment by duderino** [comment]
@ostromart @esnible I don't understand what should happen here.  Please describe.  For now, I think this is not a P0 but will put P1 so we revisit it again for the 1.1.x series

**Comment by esnible** [comment]
@ayj Is it sufficient to add these to the deprecatedFlags we already have in _istioctl/cmd/istioctl/kubeinject.go_?

```
	deprecatedFlags := []string{"coreDump", "imagePullPolicy", "includeIPRanges", "excludeIPRanges", "hub", "tag",
		"includeInboundPorts", "excludeInboundPorts", "debug", "verbosity", "sidecarProxyUID", "setVersionString"}
```

If no one is using them it should be sufficient just to mark them as deprecated so they don't appear in the docs.

Alternately I can remove them all together, or try to get them working.  Your call.

**Comment by ostromart** [comment]
Yes, I think deprecating these is the way to go. If you're going to do that @esnible it would also be worth documenting the env var behavior that Jason mentioned somewhere in the main cmd file if that's not already there.

**Comment by esnible** [comment]
https://github.com/istio/istio/pull/10830 provides an alternate PR for this problem.

I hope one of them will be merged so that the docs will not discuss non-working flags.

**Comment by ayj** [comment]
#10830 targets master. For 1.1 at most we may want to simply mark these flags as deprecated. It may be too late for 1.1.0 but could be included in 1.1.1.
